homeassistant/components/pglab/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""PG LAB Electronics integration."""
+from __future__ import annotations
+
+from pypglab.mqtt import Client, Sub_State, Subcribe_CallBack
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.components.mqtt.subscription import (
+    async_prepare_subscribe_topics,
+    async_subscribe_topics,
+    async_unsubscribe_topics,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    DEVICE_ALREADY_DISCOVERED,
+    DISCONNECT_COMPONENT,
+    DISCOVERY_INSTANCE,
+    PLATFORMS,
+)
+from .discovery import CreateDiscovery
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up PG LAB Electronics integration from a config entry."""
+
+    # define the call back for pglab  module to publish a mqtt message
+    async def mqtt_publish(
+        topic: str, payload: str, qos: int | None = 0, retain: bool | None = False
```

Comment:
Why does this need default values for qos and retain, doesn't `pypglab` call it with all parameters defined?

Reply:
Actually there is no need for the default value. I'm going to remove it